modules/accounts/email.nix

```diff
@@ -6,6 +6,45 @@ let
 
   cfg = config.accounts.email;
 
+  gpgModule = types.submodule {
+    options = {
+      gpgKey = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = "Path to your gpg key.";
+      };
+
+      onSend = mkOption {
+        type = types.enum [ "sign" "attach" "nothing" ];
+        default = "nothing";
+        description = "Method to communicate the signature.";
+      };
+    };
+  };
+
+  signatureModule = types.submodule {
+    options = {
+      content = mkOption {
+        type = types.str;
+        default = "";
+        example = ''
+          --
+          Luke Skywalker
+          May the force be with you.
+        '';
+        description = ''
+          Signature content. You can use builtins.readFile to load it from a file.
+        '';
+      };
+
+      showSignature = mkOption {
```

Comment:
I'm not sure about the support of this option under different clients. Do you know which ones will support this?

Reply:
I believe it's widely available. 
For thunderbird
https://www.google.com/imgres?imgurl=https://www.mail-signatures.com/wp-content/uploads/2015/08/html-text-modif.png&imgrefurl=https://www.mail-signatures.com/articles/how-to-set-up-email-signature-in-thunderbird/&h=699&w=646&tbnid=F2mywyKtBKGCpM:&q=attach+signature+thunderbird&tbnh=160&tbnw=147&usg=AFrqEzeSx3rcEwSBUx7Jk1lFZ-p0vegbHg&vet=12ahUKEwiRhsanta_dAhVKOrwKHSlaCSAQ9QEwAHoECAcQBg..i&docid=tAaJc1qXq6_NTM&client=firefox-b&sa=X&ved=2ahUKEwiRhsanta_dAhVKOrwKHSlaCSAQ9QEwAHoECAcQBg

astroid supports it too
https://github.com/astroidmail/astroid/wiki/Configuration-Reference
accounts.[accountname].signature_attach

Along with alot.

mutt just supports attaching a signature I believe. I could not find for evolution.